src/scripts/OSFramework/OSUI/Pattern/TimePicker/scss/_timepicker.scss

```diff
@@ -34,9 +34,9 @@
 	}
 
 	// Hide the platform input which is set as hidden by the library and we're change it into the expected type, however we do not want it visible since library will add a clone to better deal with the selected dates.
-	& > span > input {
+	& span > input.flatpickr-input {
 		&:first-of-type {
```

Comment:
here the same: ".flatpickr-input:first-of-type"

Reply:
@JoaoFerreira-FrontEnd In this case that is what we want. We always want to keep the input hidded in all use cases. This input is hidden and is cloned for the datepicker.

Reply:
I did understand the usecase, what I've tried to share is that here my suggestion is the some as the above, but needs to be a bit more complex, because we have other rule in the file that will be stronger than ".flatpickr-input:first-of-type". that's why I've suggest... "input.flatpickr-input:first-of-type"

Reply:
The Datepicker needs to ha 2 inputs, one from the provider and one from that platform. One of them is cloned and we need to hide the other one from DOM. This is how we have the datepicker implemented and can match the date from provider and platform. The use case is, the hidden input is being show in some use cases because the selector on OSUI is added only for the direct child. If the user adds a custom content or wrap the inpout with the container, the selector will not work.

Tha being said, we need to be more specific and remove the !important from the selector. So all the inputs with the class .flatpickr-input will be hidden and only the first one, that we controls. If the users want to add the flatpickr class for some reason, we will not hide the element.